pallets/pool-fees/src/lib.rs

```diff
@@ -0,0 +1,840 @@
+
+		/// The source of truth for pool permissions.
+		type Permissions: Permissions<
+			Self::AccountId,
+			Scope = PermissionScope<Self::PoolId, Self::CurrencyId>,
+			Role = Role<Self::TrancheId>,
+			Error = DispatchError,
+		>;
+
+		/// The pool fee bound per bucket. If multiplied with the number of
+		/// bucket variants, this yields the max number of fees per pool.
+		type MaxPoolFeesPerBucket: Get<u32>;
+
+		// TODO: Should be MaxPoolFeesPerBucket * PoolFeeBucket::iter().count() as u32
+		type MaxFeesPerPool: Get<u32>;
+
+		/// Identifier of this pallet used as an account which temporarily
+		/// stores disbursing fees in between closing and executing an epoch.
+		#[pallet::constant]
+		type PalletId: Get<PalletId>;
+
+		/// Fetching method for the time of the current block
+		type Time: TimeAsSecs;
+
+		/// The provider for the positive pool NAV on which pool fees are
+		/// dependent.
+		type PositiveNAV: PoolNAV<Self::PoolId, Self::Balance>;
+
+		/// The maximum age of the positive NAV.
+		///
+		/// NOTE: Temporary solution until accounting pallet.s
+		type MaxAgePositiveNAV: Get<Seconds>;
+
+		// TODO: Enable after creating benchmarks
+		// type WeightInfo: WeightInfo;
+	}
+
+	/// Maps a pool to their corresponding fee ids with [PoolFeeBucket]
+	/// granularity.
+	///
+	/// Lifetime of a storage entry: Forever, inherited from pool lifetime.
+	#[pallet::storage]
+	pub type FeeIds<T: Config> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		T::PoolId,
+		Blake2_128Concat,
+		PoolFeeBucket,
+		BoundedVec<T::FeeId, T::MaxPoolFeesPerBucket>,
+		ValueQuery,
+	>;
+
+	/// Source of truth for the last created fee identifier.
+	///
+	/// Lifetime: Forever.
+	#[pallet::storage]
+	pub type LastFeeId<T: Config> = StorageValue<_, T::FeeId, ValueQuery>;
+
+	/// Maps a fee identifier to the corresponding pool and [PoolFeeBucket].
+	///
+	/// Lifetime of a storage entry: Forever, inherited from pool lifetime.
+	#[pallet::storage]
+	pub type FeeIdsToPoolBucket<T: Config> =
+		StorageMap<_, Blake2_128Concat, T::FeeId, (T::PoolId, PoolFeeBucket), OptionQuery>;
+
+	/// Represents the active fees for a given pool id and fee bucket. For each
+	/// fee, the limit as well as pending, disbursement and payable amounts are
+	/// included.
+	///
+	/// Lifetime of a storage entry: Forever, inherited from pool lifetime.
+	#[pallet::storage]
+	pub type ActiveFees<T: Config> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		T::PoolId,
+		Blake2_128Concat,
+		PoolFeeBucket,
+		BoundedVec<PendingPoolFeeOf<T>, T::MaxPoolFeesPerBucket>,
+		ValueQuery,
+	>;
+
+	/// Stores the (negative) portfolio valuation associated to each pool
+	/// derived from the pending fee amounts.
+	///
+	/// Lifetime of a storage entry: Forever, inherited from pool lifetime.
+	#[pallet::storage]
+	pub(crate) type PortfolioValuation<T: Config> = StorageMap<
+		_,
+		Blake2_128Concat,
+		T::PoolId,
+		portfolio::PortfolioValuation<T::Balance, T::FeeId, T::MaxFeesPerPool>,
```

Comment:
Pool fees only care about the `portfolio.value` and `portfolio.last_updated`. However, I am unsure whether for accounting the values must be there as well. In the end, all of this will be refactored with the accounting pallet anyways. My assumption was that it's easier to refactor if both portfolio updating pallets used the same logic.

Reply:
Mmm... I see. Maybe it would make sense to have the portfolio through a trait? I mean, the current `PoolNav` trait could return an object that implements the `Portfolio` trait, which could then give access to the value and `last_updated` fields. Just a thought!